response_operations_ui/views/messages.py

```diff
@@ -195,6 +199,39 @@ def view_selected_survey(selected_survey):
                                response_error=True)
 
 
+@messages_bp.route('/threads/<thread_id>/close-conversation', methods=['GET', 'POST'])
+@login_required
+def close_conversation(thread_id):
+    thread_conversation = message_controllers.get_conversation(thread_id)
+    refined_thread = [_refine(message) for message in reversed(thread_conversation['messages'])]
+    try:
+        selected_survey = session['messages_survey_selection']
+    except KeyError:
+        logger.exception('Session unavailable')
+        selected_survey = None
+
+    if request.method == 'POST':
+        if request.args.get('reopen_conversation'):
+            message_controllers.update_close_conversation_label(thread_id=thread_id, status=False)
```

Comment:
If we're re-opening a conversation we don't need to have loaded the full conversation, so we should try and do this before loading all of the messages.